cloud_experts_tutorials/rosa-mobb-dynamic-certificate-custom-domain-tutorial.adoc

```diff
@@ -0,0 +1,531 @@
+:_content-type: ASSEMBLY
+[id="rosa-mobb-dynamic-certificate-custom-domain-tutorial"]
+= Tutorial: Dynamically issuing certificates with a custom domain in ROSA
+include::_attributes/attributes-openshift-dedicated.adoc[]
+:context: rosa-mobb-dynamic-certificate-custom-domain-tutorial
+
+toc::[]
+
+//Mobb content metadata
+//Brought into ROSA product docs 2023-09-20
+//---
+//date: '2022-10-11'
+//title: Dynamic Certificates for ROSA Custom Domain
+//weight: 1
+//tags: ["AWS", "ROSA"]
+//authors:
+//   Kevin Collins
+//---
+
+There may be situations when you prefer not to use `wild-card` certificates. This Red Hat OpenShift Service on AWS (ROSA) guide talks about certificate management with cert-manager and letsencrypt, to dynamically issue certificates to routes created on a custom domain that's hosted on AWS Route53.
+
+== Prerequisites
+
+* A ROSA cluster
+* You are logged into the `oc` CLI
+* You are logged into the AWS CLI
+* You are logged into the `rosa` CLI
+* jq
+* gettext
+* A public Route53 hosted zone, and the related domain to use
+
+== Set up environment
+
+. Export a few environment variables:
++
+[source,terminal]
+----
+   export CLUSTER_NAME="sts-pvtlnk-cluster"
+   export OIDC_PROVIDER=$(oc get authentication.config.openshift.io cluster -o json \
+   | jq -r .spec.serviceAccountIssuer| sed -e "s/^https:\/\///")
+   export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+   export SCRATCH_DIR=/tmp/scratch
+   export AWS_PAGER=""
+   export LETSENCRYPT_EMAIL=youremail@work.com
+   export HOSTED_ZONE_ID=ABCDEFGHEXAMPLE
+   export HOSTED_ZONE_REGION=us-east-2
+   export DOMAIN=lab.domain.com   #Custom Hosted Zone Domain for apps
+   mkdir -p $SCRATCH_DIR
+----
+
+. Install jq & gettext. Installing or ensuring the gettext & jq package is installed allows you to use envsubst to simplify some of your configuration. You can then use the output of the CLI commands as the input into YAML files to reduce the complexity of manual editing.
++
+[source,terminal]
+----
+$ brew install gettext jq
+----
++
+For Linux or Windows WSL, use the following command:
++
+[source,terminal]
+----
+$ sudo dnf install gettext jq
+----
+
+== Prepare the AWS account
+In order to make changes to the AWS Route53 hosted zone to add or remove DNS TXT challenge records by the cert-manager pod, you first need to create an IAM role with specific policy permissions and a trust relationship to allow access to the pod.
+
+The custom domain hosted zone that is used in this guide is in the same account as the ROSA cluster. If these are in different accounts, a few additional steps for link:https://cert-manager.io/docs/configuration/acme/dns01/route53/#cross-account-access[Cross Account Access] is required.
+
+. Prepare an IAM Policy file
++
+[source,json]
+----
+$   cat <<EOF > $SCRATCH_DIR/cert-manager-r53-policy.json
```

Comment:
Not sure if it's just the netlify preview, but it looks like this is all intended too far - check out the difference between what gets copied from the original https://cloud.redhat.com/experts/rosa/dynamic-certificates/ versus the netlify preview: https://65007--docspreview.netlify.app/openshift-rosa/latest/cloud_experts_tutorials/rosa-mobb-dynamic-certificate-custom-domain-tutorial